### PR TITLE
refactor(nervous-system): Remove ic-sns-governance and ic-sns-init's test_feature configurations

### PR DESCRIPTION
**Pull request by anchpop**
[← Previous PR](https://github.com/dfinity/ic/pull/1609)

There are two reasons to have a --test_feature target:

1. The crate itself has different functionality when cfg(feature = "test") is true
2. The crate is a library that depends on crate that has a --test_feature target, and needs to give the choice to its consumer whether the dependency has --test_feature enabled.

The second rule is "viral" which has lead to a massive outbreak of crates with a `--test_feature` target. In fact most of our crates that had a --test_feature target were in the second category.

The root cause of most of these was that SNS Governance's library had a `--test_feature` target. This PR:

1. Moves the logic for determining whether we are in "test mode" to the canister crate (from the library crate)
2. Remove's SNS Governance's library's --test-feature target
3. Remove's SNS Init's --test-feature target, which did nothing after #1609
4. Reverses the viral outbreak of --test-feature now that these two crates no longer needed it. The following crates have had their --test-feature removed entirely:
   1. `//rs/sns/init:init--test_feature`
   2. `//rs/sns/governance:governance--test_feature`
   4. `//rs/nns/governance/api:api--test_feature`
   5. `//rs/nns/sns-wasm:sns-wasm--test_feature`
   6.  `//rs/sns/cli:cli--test_feature`
   7.  `//rs/sns/swap:swap--test_feature`
   8.  `//rs/nns/governance/init:init--test_feature`
   8.  `//rs/nns/gtc:gtc--test_feature`

And some others no longer depend on any --test_feature crates, which is useful for development

### CODE REVIEW

**Review by nmattia — APPROVED** [review]
Rubber stamp

File: rs/sns/governance/src/governance/tests/fail_stuck_upgrade_in_progress_tests.rs [hunk 1]
```diff
@@ -114,7 +114,8 @@ fn test_does_nothing_if_there_is_no_upgrade_in_progress() {
         Box::new(DoNothingLedger {}),
         Box::new(DoNothingLedger {}),
         Box::new(FakeCmc::new()),
-    );
+    )
+    .enable_test_features();
```
Comment (max-dfinity) [inline rs/sns/governance/src/governance/tests/fail_stuck_upgrade_in_progress_tests.rs:118]: Why is this needed in these tests?  Seems a bit surprising.

Comment (anchpop) [reply rs/sns/governance/src/governance/tests/fail_stuck_upgrade_in_progress_tests.rs]: I removed it!
